presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -118,10 +119,13 @@ public ConnectorPageSource createPageSource(ConnectorTransactionHandle transacti
             List<HiveColumnHandle> hiveColumns,
             List<HivePartitionKey> partitionKeys,
             DateTimeZone hiveStorageTimeZone,
-            TypeManager typeManager)
+            TypeManager typeManager,
+            Map<Integer, HiveType> columnCoercions)
     {
-        List<ColumnMapping> columnMappings = ColumnMapping.buildColumnMappings(partitionKeys, hiveColumns, path);
-        List<HiveColumnHandle> regularColumns = ColumnMapping.extractRegularColumns(columnMappings);
+        List<ColumnMapping> columnMappings = ColumnMapping.buildColumnMappings(partitionKeys, hiveColumns, columnCoercions, path);
+        List<ColumnMapping> regularColumnMappings = ColumnMapping.extractRegularColumnMappings(columnMappings);
+        List<HiveColumnHandle> regularOriginalColumnHandles = ColumnMapping.extractRegularColumnHandles(regularColumnMappings, false);
+        List<HiveColumnHandle> regularCoercedColumnHandles = ColumnMapping.extractRegularColumnHandles(regularColumnMappings, true);
```

Comment:
I find these two variables hard to understand and I would inline them as noted below.